src/shogun/lib/SGMatrix.h

```diff
@@ -306,9 +309,44 @@ template<class T> class SGMatrix : public SGReferencedData
 		/** Set matrix to a constant */
 		void set_const(T const_elem);
 
-		/** fill matrix with zeros */
+		/** Fill matrix with zeros */
 		void zero();
 
+		/**
+		 * Fill matrix with uniformly distributed randoms
+		 * Float Types: [0, 1)
+		 * Int Types: std::numeric_limit<T>::(min->max)
+		**/
+		template <typename U = T> requires Floating<U> && RngVectorizable<U>
+		void random() {
+			auto r = std::make_unique<CRandom>();
+			r->fill_array_co(matrix, num_rows * num_cols);
+		}
+
+		template <typename U = T> requires Integral<U> && RngVectorizable<U>
```

Comment:
Yes! there is - I was hitting a slight problem when trying to set it up though because of the explicit specializations across all types that have been defined. I plan on breaking down the concepts one level more to:
1. Integral/Floating
2. Rngable
3. RngVectorizable
This should ensure that only correct matches occur.

Reply:
Looking forward to see it :-)

Reply:
This is turning out way harder than I imagined :/
So the thing is concepts dont interfere with name mangling - But this also means that during explicit specialization to the compiler it looks like there are two duplicate templated member functions. Here is a minimal reconstruction: https://coliru.stacked-crooked.com/a/9d472eef7579adc9

Reply:
That seems interesting indeed. I wonder whether this is actually expected behavior or some (potentially current) limitation or bug. It could be worth opening a conversation about it, if there's none yet (just an e-mail should do or even a short blog post illustrating the issue).

For the purpose of this pull request, I say we keep it as you solved the problem. Very nice job debugging it and explaining here, thank you!